pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1461,10 +1461,11 @@ public PinotResourceManagerResponse dropTable(String tableName) {
   }
 
   public void addNewSegment(@Nonnull SegmentMetadata segmentMetadata, @Nonnull String downloadUrl) {
-    addNewSegment(segmentMetadata, downloadUrl, null);
+    addNewSegment(segmentMetadata, downloadUrl, null, null);
```

Comment:
get the assigned instances here before calling the new API?

Reply:
Added.